torch/csrc/jit/passes/peephole.cpp

```diff
@@ -41,6 +41,20 @@ void PeepholeOptimize(Block * block) {
           // Let DCE clean up any unused nodes at this point
         }
       } break;
+      case aten::type_as: {
+        JIT_ASSERT(n->inputs().size() == 2);
+        Value* LHS(n->input(0));
+        Value* RHS(n->input(1));
+        // If LHS and RHS have the same static type, remove the type_as operator.
+        if (RHS->type()->kind() == TypeKind::TensorType) {
+           auto LType = (*LHS->type()).cast<TensorType>();
```

Comment:
The `cast` function returns null when something is not a tensor type, but is not being checked for the LHS here. You want something like:

```
if(TensorType* lhs = n->input(0)->type()->cast<TenorType>()) {
  if(TensorType* rhs = n->input(1)->type()->cast<TenorType>()) {
     ...
```